workers/generic-worker/main.go

```diff
@@ -584,52 +583,25 @@ func ClaimWork() *TaskRun {
 
 func (task *TaskRun) validatePayload() *CommandExecutionError {
 	jsonPayload := task.Definition.Payload
-	defaults.SetDefaults(&task.Payload)
-	err := json.Unmarshal(jsonPayload, &task.Payload)
-	if err != nil {
-		return MalformedPayloadError(err)
+	validateErr := task.validateJSON(jsonPayload, JSONSchema())
+	if validateErr != nil {
+		return validateErr
 	}
-	schemaLoader := gojsonschema.NewStringLoader(JSONSchema())
-	docLoader := gojsonschema.NewStringLoader(string(jsonPayload))
-	result, err := gojsonschema.Validate(schemaLoader, docLoader)
+	payload := map[string]interface{}{}
+	err := json.Unmarshal(jsonPayload, &payload)
 	if err != nil {
 		return MalformedPayloadError(err)
 	}
-	if !result.Valid() {
-		task.Errorf("Task payload for this worker type must conform to the following jsonschema:\n%s", JSONSchema())
-		task.Error("TASK FAIL since the task payload is invalid. See errors:")
-		for _, desc := range result.Errors() {
-			task.Errorf("- %s", desc)
+	if _, exists := payload["image"]; exists {
+		err := task.convertDockerWorkerPayload()
+		if err != nil {
+			return err
+		}
+	} else {
+		err := json.Unmarshal(jsonPayload, &task.Payload)
+		if err != nil {
+			return MalformedPayloadError(err)
```

Comment:
again, this should probably panic here if there is an error. we have already seen that it passes json schema validation, so not being able to unmarshal is pretty serious.